pkg/channel/consolidated/dispatcher/dispatcher.go

```diff
@@ -253,6 +254,44 @@ func (d *KafkaDispatcher) UpdateHostToChannelMap(config *Config) error {
 	return nil
 }
 
+func (d *KafkaDispatcher) CleanupChannel(name, namespace, hostname string) error {
+	channelRef := eventingchannels.ChannelReference{
+		Name:      name,
+		Namespace: namespace,
+	}
+
+	// Remove from the hostToChannel map the mapping with this channel
+	// TODO why we have a lock inside another lock for this map?!
+	d.hostToChannelMapLock.Lock()
```

Comment:
nit: I started thinking that relying on defer to guarantee unlock takes place even in panics is more robust? but I still don't have a super strong opinion about it. so I'd use an anonymous function for short scope locking here. wdyt?

Reply:
but this lock shouldn't be here at all :smile: set/get hostToChannelMap already uses a CAS pointer

Reply:
yeah....we don't speak about that though :smile: just saying in case another one tries to acquire it and this cleanup goroutine panics, the lock will stay locked IIUC.